package/MDAnalysis/core/selection.py

```diff
@@ -655,8 +655,16 @@ def _apply(self, group):
         pattern = Chem.MolFromSmarts(self.pattern)
         if not pattern:
             raise ValueError(f"{self.pattern!r} is not a valid SMARTS query")
+        if "maxMatches" in self.rdkit_kwargs:
+            max_matches = self.rdkit_kwargs.pop("maxMatches")
+        else:
+            max_matches = group.n_atoms
```

Comment:
Could we use the default from pop instead here?
```suggestion
        max_matches = self.rdkit_kwargs.pop("maxMatches", group.n_atoms)
```

Reply:
I was unfamiliar with this syntax. Way cleaner! Thanks!